x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -237,14 +244,29 @@ private void estimateMemoryUsageAndUpdateMemoryTracker(StartContext startContext
             listener::onFailure
         );
 
-        PutDataFrameAnalyticsAction.Request explainRequest = new PutDataFrameAnalyticsAction.Request(startContext.config);
-        ClientHelper.executeAsyncWithOrigin(
-            client,
-            ClientHelper.ML_ORIGIN,
-            ExplainDataFrameAnalyticsAction.INSTANCE,
-            explainRequest,
-            explainListener);
-
+        Optional<DiscoveryNode> node = findMlNode(clusterService.state());
```

Comment:
Is there some `common` package the `findMlNode` method could go to?

Reply:
I am thinking of just changing the logic so that we always call `_explain` and `_explain` handles this.